armi/materials/material.py

```diff
@@ -32,16 +33,29 @@
 from armi import runLog
 from armi.materials import materialParameters
 from armi.nucDirectory import nuclideBases
-from armi.reactor import composites
 from armi.reactor.flags import TypeSpec
+from armi.reactor.parameters import resolveCollections
 from armi.utils import densityTools
 from armi.utils.units import getTk, getTc
 
 # globals
 FAIL_ON_RANGE = False
 
 
-class Material(composites.Composite):
+class MaterialMetaType(resolveCollections.ResolveParametersMeta):
+    """
+    Metaclass for automatically defining ParameterCollection classes.
```

Comment:
Would it make more sense to not use parameters at all for this? Like, just use private variables on the materials that have some extra attributes?

If not, we need to add some more implementation documentation here explaining.

* Even though Material is not an ArmiObject, it still has params
* Are these params stateful? Do they make it into and out of the db?
* What guidelines to devs should we put here?

Reply:
Please see my question/note below about nuclides.